doc/md/writing-motoko/incomplete-code.md

```diff
@@ -0,0 +1,48 @@
+---
+sidebar_position: 27
+---
+# Writing incomplete code
+
+## Overview
+
+In the midst of writing a program, you may want to run an incomplete version or a version where one or more execution paths are either missing or simply invalid.
+
+To accommodate these situations, Motoko uses the `xxx`, `nyi` and `unreachable` functions from the base `Prelude` library, explained below. Each is a simple wrapper around a more [general trap mechanism](traps.md).
+
+## Short-term holes
+
+Short-term holes are never committed to a source repository and only ever exist in a single development session for a developer that is still writing the program.
+
+Assuming that one has imported the prelude as follows:
+
+``` motoko name=prelude
+import P "mo:base/Prelude";
+```
+
+The developer can fill any missing expression with the following:
+
+``` motoko include=prelude
+P.xxx()
+```
+
+The result will always type check at compile time, and always trap at run time, if and when this expression executes.
+
+## Long-term holes
+
+By convention, longer-term holes can be considered "not yet implemented" (`nyi`) features, and marked as such with a similar function from the Prelude module:
+
+``` motoko include=prelude
+P.nyi()
+```
+
```

Comment:
Like `xxx()`, `nyi()` has type `None`, thus any other type, and will trap on execution.